packages/mcdoc-cli/src/generate/index.ts

```diff
@@ -20,870 +20,222 @@ export async function generate(
 	project_path: string,
 	generated_path: string,
 	args: Args,
-	doc_file: walk.Item,
 	service: Service,
 	logger: Logger,
 ) {
-	const symbols = []
-
-	let errors = 0
-
-	const internal_locales: Record<string, string[]> = {}
+	// this should be a `let` if its actually used
+	const errors = 0
 
 	const locales: Record<string, string> = {}
 
-	const DocumentUri = pathToFileURL(doc_file.path).toString()
-
-	const doc_contents = await fs.readFile(doc_file.path, 'utf-8')
-
-	await service.project.onDidOpen(
-		DocumentUri,
-		'mcdoc',
-		0,
-		doc_contents,
-	)
-
-	const check = await service.project.ensureClientManagedChecked(
-		DocumentUri,
-	)
-
-	if (check && check.doc && check.node) {
-		const { doc, node } = check
-
-		const path = parse(fileURLToPath(doc.uri))
-
-		let resource = join(
-			path.dir.replace(`${project_path}`, ''),
-			path.name,
-		).replace(/^[\/\\]/, '')
-
-		// remove windows cruft
-		if (resource.includes('\\')) resource = resource.replaceAll('\\', '/')
-
-		logger.info(`parsing ${resource}\n`)
-
-		if (node.children[0]) {
-			const children = node.children
-
-			function flattenChild(
-				parent: string,
-				self: string,
-				_parent: Partial<AstNode> | undefined,
-				_child: Partial<AstNode>,
-			) {
-				const child: any = {}
-
-				const known_error = false
-
-				/* @ts-ignore */
-				child.self = self
-
-				/* @ts-ignore */
-				if (_child.parent) child.parent = parent
-
-				/* @ts-ignore */
-				if (_child.parentMap) child.parentMap = parent
-
-				if (_child.children) {
-					child.children = []
-					for (
-						const [i, __child] of Object.entries(
-							_child.children,
-						)
-					) {
-						/* @ts-ignore */
-						child.children[Number(i)] = flattenChild(
-							self,
-							`${self}[${i}]`,
-							_child,
-							__child,
-						)
-					}
-				}
-
-				child.type = _child.type
-
-				if (child.type === 'resource_location') {
-					/* @ts-ignore */
-					child.namespace = _child.namespace
-					/* @ts-ignore */
-					child.path = _child.path
-				}
-
-				if (Object.hasOwn(_child, 'isOptional')) {
+	const project_path_uri = pathToFileURL(project_path).toString() + '/'
```

Comment:
why is everything in this file snake_case 😭

Reply:
`mcdoc-cli` is my project, I'm going to keep it like this for now, we can discuss this another time like in v4.1